src/cobra/manipulation/delete.py

```diff
@@ -57,113 +59,35 @@ def prune_unused_reactions(model: "Model") -> Tuple["Model", List["Reaction"]]:
-        objects as keys.
-
-    .. deprecated::
-        Internal function that has outlived its purpose.
-
-    """
-    warn(
-        "The function `get_compiled_gene_reaction_rules` has outlived its purpose. "
-        "It will be removed soon.",
-        DeprecationWarning,
-    )
-    return {r: r.gpr for r in model.reactions}
-
-
-def find_gene_knockout_reactions(
+def knock_out_model_genes(
     model: "Model",
-    gene_list: List["Gene"],
-    compiled_gene_reaction_rules: Optional[Dict["Reaction", Module]] = None,
+    gene_list: Iterable[Union["Gene", int, str]],
 ) -> List["Reaction"]:
-    """Identify reactions which will be disabled when genes are knocked out.
+    """Disable the genes in `gene_list`.
+
+    It sets the bounds to "zero" for reactions catalysed by the genes in
+    `gene_list` if deleting the genes would stop the reactions from proceeding.
+
+    The changes are reverted upon exit if executed within the model as context.
 
     Parameters
     ----------
     model: cobra.Model
-        The model for which to find gene knock-out reactions.
+        The model whose reaction bounds are to be set.
     gene_list: list of cobra.Gene
         The list of genes to knock-out.
-    compiled_gene_reaction_rules: dict of {reaction: compiled_string},
-                                  optional
-        If provided, this gives pre-compiled gene-reaction rule strings.
-        The compiled rule strings can be evaluated much faster. If a rule
-        is not provided, the regular expression evaluation will be used.
-        Because not all gene-reaction rule strings can be evaluated, this
-        dict must exclude any rules which can not be used with eval
-        (default None).
 
     Returns
     -------
-    list of cobra.Reaction
-       The list of cobra.Reaction objects which will be disabled.
-
-    .. deprecated:: 0.22.1
-        Internal function that has outlived its purpose.
-
+    list[cobra.Reaction]
+        A list of cobra.Reactions that had the bounds turned to zero.
     """
-    warn(
-        "The function `find_gene_knockout_reactions` has outlived its purpose. "
-        "It will be removed in the next minor version (0.23.0).",
-        DeprecationWarning,
-    )
-    potential_reactions = set()
-    for gene in gene_list:
-        if isinstance(gene, str):
-            gene = model.genes.get_by_id(gene)
-        potential_reactions.update(gene._reaction)
-    gene_set = {str(i) for i in gene_list}
-    if compiled_gene_reaction_rules is None:
-        compiled_gene_reaction_rules = {r: r.gpr for r in potential_reactions}
-
-    return [
-        r
-        for r in potential_reactions
-        if not compiled_gene_reaction_rules[r].eval(gene_set)
-    ]
+    gene_list = model.genes.get_by_any(gene_list)
+    rxn_set = set()
+    for gene in model.genes.get_by_any(gene_list):
+        gene.knock_out()
+        rxn_set.update(gene.reactions)
+    return [rxn for rxn in list(rxn_set) if not rxn.functional]
```

Comment:
Converting the set to a list should not be necessary since you can iterate over a set just fine.